modifiers/allocation/modifier.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+from enum import Enum
+from ramble.modkit import *
+
+
+class AllocOpt(Enum):
+    # Experiment resource requests
+    N_RANKS = 1
+    N_NODES = 2
+    N_CORES_PER_TASK = 3
+    N_THREADS = 4  # number of OMP threads per task
+    N_RANKS_PER_NODE = 5
+    N_GPUS = 6
+
+    # Descriptions of resources available on systems
+    GPUS_PER_NODE = 100
+    CPUS_PER_NODE = 101
+
+    # Scheduler identification and other high-level instructions
+    SCHEDULER = 200
+    TIMEOUT = 201  # This is assumed to be in minutes
```

Comment:
the scheduler needs a few more pieces of info:
- batch_time
- batch queue (we have pdebug and pbatch usually, but other centers may name these different.  Often, special batches are created to give priority to a project etc.)
- account/bank (a bank/allocation we want to 'charge' for what we are working on - which is a time allocation for our project.  This is common across centers)

Reply:
* I used the more generic term "timeout" to account for "batch_time". Right now, this can specify a value in minutes in any experiment/system config file, and the modifier can map that to an appropriate request format on slurm/flux/etc.
* Agree
* Agree

For the last two, we didn't have that before and I haven't added it yet in this PR but I think it would be straightforward to do.